src/main/kotlin/JobState.kt

```diff
@@ -0,0 +1,64 @@
+package franz.internal
+
+fun <T, U> JobDSL<T, U>.asPipe(): JobState<U> = JobState(this.value)
+
+class JobState<U>(val value: U){
+    var status: JobStatus = JobStatus.Incomplete
+
+        get() = field
+
+        set(value) {
+            if(status == JobStatus.Incomplete)
```

Comment:
1. What should throw? The getter? It cannot be private, because then the _status_ property cannot be accessed from outside the class.

2. Because if we do
```
val state: JobState = job.asPipe()
    .validate { doSomethingThatWorks }
    .validate { doSomethingThatWorks }

(some intermediary operation)

state
    .validate { doSomethingThatWorks }
    .validate { doSomethingThatFails }

(some intermediary operation again)

state
    .validate { doSomething } // <-- this should not be possible, but it is if state is only passed 
// down through chaining
```

Currently, no calls are copy constructor?